include/zephyr/bluetooth/audio/bap.h

```diff
@@ -1019,6 +1019,19 @@ struct bt_bap_unicast_server_cb {
 	int (*release)(struct bt_bap_stream *stream, struct bt_bap_ascs_rsp *rsp);
 };
 
+/**
+ * @brief Configures the number of ASEs to be used for a direction.
+ *
+ * Has to be called before any call to bt_bap_unicast_server_register_cb.
+ *
+ * @param dir   Direction of the ASE.
+ * @param count Number of ASEs to use. Has to be between 0 and
+ *              CONFIG_BT_ASCS_ASE_SNK_COUNT/CONFIG_BT_ASCS_ASE_SRC_COUNT
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+int bt_bap_unicast_server_set_ase_count(enum bt_audio_dir dir, uint8_t count);
```

Comment:
Did you consider the approach of implementing two functions:

`bt_bap_unicast_server_register(uint8_t src_cnt, uint8_t snk_cnt)` that registers ASCS with the provided number of ASEs
`bt_bap_unicast_server_unregister` that unregisters ASCS

That way it's very clear what the functions do, and we do not need to introduce a dependency on `bt_bap_unicast_server_register_cb`

Reply:
Resolving this comment and considering it a part of the comment above instead.